### PR TITLE
Add support for different schedule intervals formats

### PR DESCRIPTION
**Pull request by tsvetkov-ii**
Dag schedule_interval in airflow except cron expression and timedelta may be None or cron preset. 

This PR adds this two schedule_interval formats.

### CODE REVIEW

File: src/airflow_declarative/schema.py [hunk 1]
```diff
@@ -120,10 +122,13 @@ def represent_callable(self, obj):
 CLASS = Class()
 DATE = Date()
 EMAIL = Email
+NULL = Null()
 POSITIVE_INT = Int(gte=0)
 STRING = String()
 TIMEDELTA = TimeDelta()
 
+AIRFLOW_NULL = NULL | Regexp(regexp="None")
+CRON_PRESETS = Regexp(regexp=r"@\w+")
```
Comment (kxepal) [inline src/airflow_declarative/schema.py]: This actually catch a lot of things, while Airflow supports a [limited set](http://airflow.apache.org/scheduler.html#dag-runs) of those. Let's not allow our users to make a mistake.

File: src/airflow_declarative/schema.py [hunk 2]
```diff
@@ -120,10 +122,13 @@ def represent_callable(self, obj):
 CLASS = Class()
 DATE = Date()
 EMAIL = Email
+NULL = Null()
 POSITIVE_INT = Int(gte=0)
 STRING = String()
 TIMEDELTA = TimeDelta()
 
+AIRFLOW_NULL = NULL | Regexp(regexp="None")
```
Comment (kxepal) [inline src/airflow_declarative/schema.py]: I believe it should be `^None$`. And it's true that there are no way to match a static string without regexp?

Comment (KostyaEsmukov) [reply src/airflow_declarative/schema.py]: Enum seems to be a viable alternative: https://github.com/Deepwalker/trafaret/blob/40a521ae0f580a0e5a6f3168a59cbba350d5b3ba/trafaret/base.py#L1198-L1222

But I have a question: why would we want to allow `None` as string instead of accepting yaml's `null` which would be turned to the Python's `None` by yaml parser?

File: src/airflow_declarative/schema.py (hunk 1 above)
Comment (KostyaEsmukov) [reply src/airflow_declarative/schema.py]: I guess Airflow's DAG should raise an error on `__init__` when an unsupported cron preset is passed. But I wouldn't rely on that either, given that they might decide to move out that validation from `__init__` like they did with dag cycles before.

So I think a better way is to just enumerate them all, maybe with trafaret's `Enum` atom. If something will be missing here (and present in Airflow), this wouldn't be a big deal -- expanding API is easier than shrinking. We already enumerate a bunch of Airflow attributes anyways.

File: src/airflow_declarative/schema.py (hunk 2 above)
Comment (tsvetkov-ii) [reply src/airflow_declarative/schema.py]: Can do it with Enum. I allow None as string, because airflow allow None and 'None'. Both this works, for this PR, but need to add test for null.

Comment (KostyaEsmukov) [reply src/airflow_declarative/schema.py]: That is a rather strange decision from the Airflow side. Anyways, we don't have to add support for `"None"`. I would suggest to allow only the YAML's `null`. If the behavior is the same as for `"None"`, that should be enough. Having less ways to achieve the same is usually better.

File: src/airflow_declarative/schema.py (hunk 1 above)
Comment (tsvetkov-ii) [reply src/airflow_declarative/schema.py]: I limit presets to listed in documentation with Enum.

File: src/airflow_declarative/schema.py (hunk 2 above)
Comment (tsvetkov-ii) [reply src/airflow_declarative/schema.py]: Drop support for 'None' as string.
